trigger-actions-framework/main/default/classes/TriggerBaseTest.cls

```diff
@@ -144,6 +144,25 @@ private class TriggerBaseTest {
 		);
 	}
 
+	@IsTest
+	private static void run_false() {
+		Exception myException;
+		Boolean validateRun = true;
+		Test.startTest();
+		base.context = System.TriggerOperation.BEFORE_INSERT;
+		base.triggerNew = new List<SObject>{ MY_ACCOUNT };
+		TriggerBase.bypass(ACCOUNT);
+		try {
+			base.run();
+			validateRun = base.validateRun();
```

Comment:
Do we need to call `run()` here in order to appropriately test that `validateRun` is false?

Reply:
By using `base.run()` we get code coverage for that one line 38 in TriggerBase (`return;`).
I know, `validateRun()` is run twice then... So, shall I remove that?

Reply:
I think that if we want to truly write a unit test for the `return;` on 38, we probably need to modify the inner `BaseTest` class, and make an actual assertion that nothing happened when the object is bypassed. We could just do it with a simple `Boolean` flag:

```java
class BaseTest extends TriggerBase implements TriggerAction.BeforeInsert, TriggerAction.AfterInsert, TriggerAction.BeforeUpdate, TriggerAction.AfterUpdate, TriggerAction.BeforeDelete, TriggerAction.AfterDelete, TriggerAction.AfterUndelete {
  public Boolean hasBeenExecuted = false;
  public void beforeInsert(List<SObject> newList) {
    this.hasBeenExecuted = true;
  }
  public void afterInsert(List<SObject> newList) {
    this.hasBeenExecuted = true;
  }
  public void beforeUpdate(List<SObject> newList, List<SObject> oldList) {
    this.hasBeenExecuted = true;
  }
  public void afterUpdate(List<SObject> newList, List<SObject> oldList) {
    this.hasBeenExecuted = true;
  }
  public void beforeDelete(List<SObject> oldList) {
    this.hasBeenExecuted = true;
  }
  public void afterDelete(List<SObject> newList) {
    this.hasBeenExecuted = true;
  }
  public void afterUndelete(List<SObject> newList) {
    this.hasBeenExecuted = true;
  }
}
```

Then we can remove the unnecessary `@TestVisible` on `validateRun()` in `TriggerBase`, and assert that `hasBeenExecuted` is populated properly:

```
@IsTest
private static void run_true() {
  Exception myException;
  base.context = System.TriggerOperation.BEFORE_INSERT;
  base.triggerNew = new List<SObject>{ MY_ACCOUNT };
  Test.startTest();
  try {
     base.run();
  } catch (Exception e) {
     myException = e;
  }
  Test.stopTest();
  System.assertEquals(null, myException);
  System.assertEquals(true, base.hasBeenExecuted);
}

@IsTest
private static void run_false() {
  Exception myException;
  TriggerBase.bypass(ACCOUNT);
  Test.startTest();
  try {
     base.run();
  } catch (Exception e) {
     myException = e;
  }
  Test.stopTest();
  System.assertEquals(null, myException);
  System.assertEquals(false, base.hasBeenExecuted);
}
```

Reply:
Great thoughts, thanks! Two points:
* For your `run_false()` part I added `base.triggerNew = new List<SObject>{ MY_ACCOUNT };` to make it work.
* I just saw that we should set MY_ACCOUNT.Id to null for before insert runs. I added this as well.